plan/predicate_push_down.go

```diff
@@ -282,16 +282,16 @@ func isNullRejected(ctx sessionctx.Context, schema *expression.Schema, expr expr
 
 // concatOnAndWhereConds concatenate ON conditions with WHERE conditions.
 func concatOnAndWhereConds(join *LogicalJoin, predicates []expression.Expression) []expression.Expression {
-	equalConds, leftConds, rightConds, otherConds := join.EqualConditions, join.LeftConditions, join.RightConditions, join.OtherConditions
-	ans := make([]expression.Expression, 0, len(equalConds)+len(leftConds)+len(rightConds)+len(predicates))
-	for _, v := range equalConds {
-		ans = append(ans, v)
+	numAllFilters := len(join.EqualConditions) + len(join.LeftConditions) + len(join.RightConditions) + len(join.OtherConditions) + len(predicates)
+	allFilters := make([]expression.Expression, 0, numAllFilters)
+	for _, equalCond := range join.EqualConditions {
+		allFilters = append(allFilters, equalCond)
```

Comment:
Why not append it like below do?

Reply:
`allFilters` is a slice of interface, `equalCond` is a slice of implementations, we can not do it in golang.